code/modules/surgery/organs/heart.dm

```diff
@@ -172,40 +172,57 @@
 	colour = "red"
 
 /obj/item/organ/heart/cybernetic
-	name = "cybernetic heart"
-	desc = "An electronic device designed to mimic the functions of an organic human heart. Also holds an emergency dose of epinephrine, used automatically after facing severe trauma."
+	name = "basic cybernetic heart"
+	desc = "A basic electronic device designed to mimic the functions of an organic human heart."
 	icon_state = "heart-c"
 	organ_flags = ORGAN_SYNTHETIC
-	maxHealth = 1.1 * STANDARD_ORGAN_THRESHOLD
 
-	var/dose_available = TRUE
+	var/dose_available = FALSE
 	var/rid = /datum/reagent/medicine/epinephrine
 	var/ramount = 10
+	var/emp_vulnerability = 80	//Chance of permanent effects if emp-ed.
+
+/obj/item/organ/heart/cybernetic/tier2
+	name = "cybernetic heart"
+	desc = "An electronic device designed to mimic the functions of an organic human heart. Also holds an emergency dose of epinephrine, used automatically after facing severe trauma."
+	icon_state = "heart-c-u"
+	maxHealth = 1.5 * STANDARD_ORGAN_THRESHOLD
+	dose_available = TRUE
+	emp_vulnerability = 40
+
+/obj/item/organ/heart/cybernetic/tier3
+	name = "upgraded cybernetic heart"
+	desc = "An electronic device designed to mimic the functions of an organic human heart. Also holds an emergency dose of epinephrine, used automatically after facing severe trauma. This upgraded model can regenerate its dose after use."
+	icon_state = "heart-c-u2"
+	maxHealth = 2 * STANDARD_ORGAN_THRESHOLD
+	dose_available = TRUE
+	emp_vulnerability = 20
 
 /obj/item/organ/heart/cybernetic/emp_act(severity)
 	. = ..()
 	if(. & EMP_PROTECT_SELF)
 		return
-	Stop()
-	addtimer(CALLBACK(src, .proc/Restart), 20/severity SECONDS)
-	damage += 100/severity
+	if(world.time > severe_cooldown) //So we cant just spam emp to kill people.
+		owner.Dizzy(10)
+		owner.losebreath += 10
+		severe_cooldown = world.time + 20 SECONDS
+	if(prob(emp_vulnerability))	//Chance of permanent effects
```

Comment:
I disagree with this change. I think it would be better if the probability changed with the severity (which changes with distance). As now, an emp-grenade will have the same chance and effect of messing up your day regardless if it is on the edge of your screen, or directly on at your feet.

Reply:
Agreed, severity has a place here.